test/Microsoft.AspNet.Razor.Runtime.Test/TagHelpers/DefaultTagHelperContentTest.cs

```diff
@@ -562,17 +544,14 @@ public void Fluent_SetContent_AppendFormat_WritesExpectedContent()
         {
             // Arrange
             var tagHelperContent = new DefaultTagHelperContent();
-            var expected = new[] { "First ", "Second Third" };
+            var expected = "First Second Third";
             var i = 0;
 
             // Act
             tagHelperContent.SetContent("First ").AppendFormat("{0} Third", "Second");
 
             // Assert
-            foreach (var value in tagHelperContent)
-            {
-                Assert.Equal(expected[i++], value);
-            }
+            Assert.Equal(expected, tagHelperContent.GetContent());
```

Comment:
No tests confirm `DefaultTagHelperContent` has anything to do with `BufferedHtmlcontent` or even that it maintains added content separately. In particular, `DefaultTagHelperContent` could concatenate all strings as content is added without breaking tests left in this class.

Put another way the `DefaultTagHelperContent.Buffer` property exists so that internal details can be checked (these are unit tests after all). But no tests use that "internal for testing" property.